src/realm/column_string_enum.cpp

```diff
@@ -162,6 +162,39 @@ void StringEnumColumn::do_move_last_over(size_t row_ndx, size_t last_row_ndx)
     move_last_over_without_updating_index(row_ndx, last_row_ndx); // Throws
 }
 
+void StringEnumColumn::swap_rows(size_t row_ndx_1, size_t row_ndx_2)
+{
+    REALM_ASSERT_3(row_ndx_1, <=, size());
+    REALM_ASSERT_3(row_ndx_2, <=, size());
+    REALM_ASSERT_DEBUG(row_ndx_1 != row_ndx_2);
+
+    StringData value_1 = get(row_ndx_1);
+    StringData value_2 = get(row_ndx_2);
+
+    if (value_1.is_null() && value_2.is_null()) {
+        return;
+    }
+
+    std::string buffer_1{value_1.data(), value_1.size()};
+    std::string buffer_2{value_2.data(), value_2.size()};
+
+    if (value_1.is_null()) {
+        set(row_ndx_2, realm::null());
+    }
+    else {
+        StringData copy{buffer_1.data(), buffer_1.size()};
```

Comment:
You can just swap the integers with IntegerColumn::get() / IntegerColumn::set() instead of copying payload